### DISCUSSION TITLE
After Creating .njk file to create subfolder, introducing a duplicate folder with same name breaks watch

### DISCUSSION THREAD

**Discussion by chrisativonne** [11ty/eleventy#2004] [Q&A]
Background: I'm migrating an existing static website into 11ty which includes a products page, and product subpages.

The sitemap looks like this:
home (index.njk)
products (products.njk)
products folder
- Product1.njk
- Product2.njk
- Product3.njk

Issue 1: I don't know how to nest/subfolder in 11ty (what is best practice?)
Issue 2: When I create products.njk to auto create the Products folder at _site, then proceed to paste an actual folder with the same name containing all of my individual product pages this seems to break watch.  The contents of "product" folder don't appear under _site, and this seems to disrupt any other changes being picked up and replicated to _site.

What am I doing wrong?

**Comment by pdehaan** [comment]
I almost never use “watch”, but are you getting an error when you run build?
If you have the site in github or somewhere public, I can try cloning and taking a look.

**Reply by chrisativonne** [reply]
I'm not sure how to move this into a Git. I tried installing the client on my mac but it wont let me commit more than one file at at a time. Whats the best way to publish the files here?

**Reply by pdehaan** [reply]
I generally go to https://github.com/new and create a new repo via GitHub, then clone that repo locally, copy my files into the cloned folder, then run some combination of `git add .` (making sure you .gitignore your ./node_modules/ folder and your generated output site folder) and then `git commit -m "changes"` and `git push origin main`. Or some rough combination of commands like that.

**Reply by chrisativonne** [reply]
Done. <redacted>

**Reply by pdehaan** [reply]
The site wasn't building locally for me due to a missing closing quote in the YAML front matter of a NJK template (see https://github.com/chrisativonne/alumiermd/pull/1) but otherwise it seems to build fine.

I can run `eleventy --serve` via a package.json script and it loads the site at http://localhost:8080/products but I might now be having a CSS loading issue since the site is unstyled locally.

Is there something specific I should be looking at?

**Reply by chrisativonne** [reply]
Strange, line 4 would be:

1 <!doctype html>
2 <html lang="en">
3   <head>
4	  <title>{{ title }}</title>
	  
What quote is missing here? [The guide I've been using here](https://www.stackbit.com/blog/jamstack-journey-templatize-static-html/) seems to have variations of quotes and unquoted

Specifically its not clear to me whether the "products" folder and its contents should automatically be creating itself in the _site folder. At some point between when I created it last night, and now, it stopped updating any changes I was making. I cannot convince sub pages to appear in the "products" folder under _site

What/Is this the right approach to build subfolders/subcontents/?

Thanks again for the going assistance.

**Reply by pdehaan** [reply]
True, there may be a conflict if you have a /products.njk (which will convert to /products/index.html by Eleventy, unless you special case the permalink) and you also have a /products/index.njk file (which will also try and write to /products/index.html). I don't think that's the case here although there is a suspicious /products/index file with no extension, but that should be ignored by Eleventy.

Another potential gotcha is that your config file is "eleventy.js" (which is fine) but by default Eleventy will look for ".eleventy.js" (with a leading period, per https://www.11ty.dev/docs/config/). If you want to use a non-default config file name, you'll need to specify the `--config` flag on the command line, per https://www.11ty.dev/docs/usage/#-config-to-change-the-config-file-name.

**Reply by pdehaan** [reply]
> What/Is this the right approach to build subfolders/subcontents/?

I think your approach is absolutely fine. I either do /products.njk or products/index.njk and then have all my products/a.njk, products/b.njk, and products/c.njk. Either way should produce the same output. The only difference can be if you're using directory data files and want your products/index page to have access to those same properties. :shrug:

**Reply by chrisativonne** [reply]
1. Ah fixed the missing "." from the eleventy.js. That definitely helped. I deleted the _site folder and then restarted eleventy. It started to build my site now.
1b) Also corrected the missing single quote. Nice catch. I thought I might have done that as I edited all of those pages individually.

2. I noticed that it doesn't automatically incorporate my "assets" folder or my "i" images folder to _site, and I need to manually add that - is this normal/expected behaviour?

**Reply by pdehaan** [reply]
> * I noticed that it doesn't automatically incorporate my "assets" folder or my "i" images folder to _site, and I need to manually add that - is this normal/expected behaviour?

Eleventy doesn’t automatically copy asset files for you. If you want to copy asset folders or files, you’ll need to use pass through copy in your config file.

**Comment by pdehaan** [comment]
Yeah, line numbers can often be very confusing. Does it count the `---` in the frontmatter as a line? are there extra linebreaks somewhere else that causes the numbers to be off? I found it mostly through trial and error and deleting all the products/*.njk files and slowly poking until I saw which front matter was bad.

The missing quote is here (you can see my PR in your repo also). Note that the single quote is opened, but never closed, so YAML doesn't know what to do and just errors out.

```yaml
title: 'Retinol Resurfacing Serum 1.0 - AlumierMD Canada
```

---

I generally avoid using `--watch` and just build my sites ad-hoc as I'm testing. Another useful tip is to just delete your generated `_site/` folder and rebuild from scratch. You could even use some node module like `del` or `rimraf` to delete the output directory from your .eleventy.js config file each time, or adding it as a "prebuild" npm script which will run before your "npm run build" script.

Sometimes you get into a case where you have a bad template, which causes a build failure which causes the templates to not be written/updated, so you're staring at an old build in `_site/` and wondering why nothing seems to be updating/working because you didn't see that some error has been breaking the builds for the past 15 minutes.

**Comment by chrisativonne** [comment]
I think this solves my immediate problem. Thanks!
